component/xload/splitter.go

```diff
@@ -264,6 +277,31 @@ func (d *downloadSplitter) Process(item *WorkItem) (int, error) {
 		return -1, fmt.Errorf("failed to download data for file %s", item.Path)
 	}
 
+	if d.consistency {
+		// Compute md5 of local file
+		fileMD5, err := common.GetMD5(item.FileHandle)
+		if err != nil {
+			log.Err("downloadSplitter::Process : Failed to generate MD5Sum for %s [%s]", item.Path, err.Error())
+		} else {
+			if item.MD5 == nil {
```

Comment:
If properties did not have md5 then there is no point in computing locally as well. We shall check d.consistency && item.MD5 != nil

Reply:
Done